Sources/NIOCore/Codec.swift

```diff
@@ -853,14 +844,25 @@ extension MessageToByteHandler {
 // MARK: ByteToMessageHandler: RemovableChannelHandler
 extension MessageToByteHandler: RemovableChannelHandler {
     public func removeHandler(context: ChannelHandlerContext, removalToken: ChannelHandlerContext.RemovalToken) {
-        do {
-            self.buffer!.clear()
-            try self.encoder.encodeLast(out: &self.buffer!)
-            context.write(self.wrapOutboundOut(self.buffer!), promise: nil)
-        } catch {
-            self.state = .error(error)
+        switch self.state {
+        case .notInChannelYet:
+            preconditionFailure("MessageToByteHandler.removeHandler called before it was added to a Channel")
+        case .error(let error):
```

Comment:
I'm unsure about this, should I still remove the handler ?

Reply:
Yes.